Makefile

```
@@ -272,9 +272,24 @@ config-generate:
 
+# The env must come from the CALLER, not from the `ENV ?= dev` default further
+# down this file. A `?=` assignment is global regardless of position, so the
+# `test -n "$(ENV)"` this replaced always saw `dev` and could never fail: a bare
+# `make config-check-drift` silently checked the dev overlay and printed
+# "✓ No drift" while staging and prod both drifted. That is exactly how #1116
+# reached CI with a `common.yaml` key that changed every component's ConfigMap.
+# `$(origin ENV)` is the only way to ask "did the caller say so?" — it reports
+# `command line` for `ENV=prod`, `file` for the default. `ENV=dev` typed
+# explicitly still works.
 .PHONY: config-check-drift
 config-check-drift:
```

checkmake (0.3.2)

[warning] Target body for "config-check-drift" exceeds allowed length of 5 lines (28).

(maxbodylength)